### PR TITLE
Extends most of water's capabilities to holy water

### LINKED ISSUE #17066: Holy water does not increase plant water level or do anything for plants
~~This goes deeper though. From what I can tell there's two entirely parallel systems for handling reagents put into plant trays.~~
EDIT: As it turns out code/game/machinery/hydroponics.dm is not included so that's egg on my face. Syringing holy water is just as useless.

https://github.com/vgstation-coders/vgstation13/blob/01b9e180f16529b1f075f56489eca99b51c92017/code/modules/hydroponics/hydroponics_reagents.dm#L42-L256

### PR DESCRIPTION
**Pull request by Pathid**
Closes #17066

:cl:
* tweak: Holy water now has most of the same capabilities water does, including firefighting, harming slimes, and watering plants. Notable things it doesn't do: slip, and act as water in reactions.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -911,6 +914,8 @@
 	if(volume >= 1)
 		O.bless()
 
+	call(/datum/reagent/water/reaction_obj)(O, volume)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Reagents.dm]: surely there's a better way than this

**Comment by JustSumBody** [comment]
If I'm cult/vamp checking does this make holy water burn greys as well?

**Comment by Pathid** [comment]
@JustSumBody Yes, it does.

**Comment by Aurx** [comment]
Holy Sacid when?

**Comment by Accipitermaris** [comment]
seems like you'd be able to play as a greyyy so sec can't waterboard you without killing you in the process if youre a suspected cultist

**Comment by NigglyWiggly** [comment]
It might be worth it to make it deal less damage to Greys, just so that you can still actually use it on them, but I'm not sure

**Comment by ghost** [comment]
Nah nah nah code holy sulphuric acid so I can be a grey chaplain

**Comment by MadmanMartian** [comment]
>"Chaplain, get to blessing the medical bay"
>chaplain sprays holy acid on medical bay
>Everything is holy, and then melts

do it

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -4001,15 +4008,7 @@
...
-	if(volume >= 3)
-		T.wet(800)
-	var/hotspot = (locate(/obj/effect/fire) in T)
-	if(hotspot)
-		var/datum/gas_mixture/lowertemp = T.remove_air(T:air:total_moles())
-		lowertemp.temperature = max( min(lowertemp.temperature-2000,lowertemp.temperature / 2), 0)
-		lowertemp.react()
-		T.assume_air(lowertemp)
-		qdel(hotspot)
+	call(/datum/reagent/water/proc/douse)(T, volume)
```
Comment (DamianX) [inline code/modules/reagents/Chemistry-Reagents.dm:4011]: This makes me uncomfortable
why is that a `/datum/reagent/water` proc anyway?

Comment (Pathid) [reply code/modules/reagents/Chemistry-Reagents.dm:4011]: Didn't mean to leave it like that. Whoops.

**Comment by DamianX** [comment]
when

**Comment by SonixApache** [comment]
WHEN

**Comment by Intigracy** [comment]
when

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 2 above)
Comment (Intigracy) [reply code/modules/reagents/Chemistry-Reagents.dm:4011]: you can use corn oil to douse fires

kek

**Comment by Pathid** [comment]
I'll work on it tomorrow, probably.

**Comment by Intigracy** [comment]
when

**Comment by Kurfursten** [comment]
it definitely happened tomorrow

**Pull request closed without merging**